azext_iot/tests/test_iot_dps_unit.py

```diff
@@ -195,32 +203,101 @@ def generate_enrollment_show(**kvp):
-        ([generate_enrollment_show(attestation = {'type': 'x509'}),
-          generate_enrollment_group_show(attestation = {'type': 'x509'})]),
+    @pytest.mark.parametrize("req", [
+        (generate_enrollment_update_req(secondary_certificate_path='someOtherCertPath')),
+        (generate_enrollment_update_req(certificate_path='newCertPath', secondary_certificate_path='someOtherCertPath')),
+        (generate_enrollment_update_req(remove_certificate_path='true')),
+        (generate_enrollment_update_req(iot_hub_host_name='someOtherHubName',
+                                        initial_twin_tags={'newKey': 'newValue'},
+                                        initial_twin_properties={'newKey': 'newValue'},
+                                        provisioning_status='enabled',
+                                        device_id='newId'))
     ])
-    def test_enrollment_group_update(self, serviceclient, result):
-        serviceclient.return_value.text = json.dumps(result)
-        subject.iot_dps_device_enrollment_update(None, enrollment_id, mock_target['entity'], resource_group, 'AAAA==', None,
-                                                 'newCertPath')
-
+    def test_enrollment_group_update(self, serviceclient, req):
+        subject.iot_dps_device_enrollment_update(None,
+                                                 req['enrollment_id'],
+                                                 req['dps_name'],
+                                                 req['rg'],
+                                                 etag,
+                                                 None,
+                                                 req['certificate_path'],
+                                                 req['secondary_certificate_path'],
+                                                 req['remove_certificate_path'],
+                                                 req['remove_secondary_certificate_path'],
+                                                 req['device_id'],
+                                                 req['iot_hub_host_name'],
+                                                 req['initial_twin_tags'],
+                                                 req['initial_twin_properties'],
+                                                 req['provisioning_status'])
+        # Index 1 is the update args
         args = serviceclient.call_args_list[1]
         url = args[0][0].url
+
         assert "{}/enrollments/{}?".format(mock_target['entity'], enrollment_id) in url
         assert args[0][0].method == 'PUT'
 
         body = args[0][2]
-'''
+
+        if not req['certificate_path']:
+            if req['remove_certificate_path']:
+                assert body['attestation']['x509']['clientCertificates']['primary'] is None
+            else:
+                assert body['attestation']['x509']['clientCertificates']['primary']['info'] is not None
+        if req['certificate_path']:
```

Comment:
You will get an index exception if ```req['thing']``` doesn't exist, better to do ```req.get('thing')``` which will return ```None``` if the index item doesn't exist.